docker/Makefile

```
@@ -14,6 +14,9 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+# PDO_SOURCE_ROOT
+PDO_SOURCE_ROOT ?= $(shell readlink -f ..)
```

This assumes make is running within the docker folder, which is not the case if you run with the `-f` option outside of it -- this does not happen to us usually, but it's a possibility.
```suggestion
mkfile_path := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
PDO_SOURCE_ROOT ?= $(realpath $(mkfile_path)../)
```
The proposed change makes the computation of the parent folder dependent on the makefile location itself.

---

Ooops, given that i usually use `make -C docker ..` i should have known better. Thanks for catching.  Pushed corresponding update merging in your changes.

---

by the way... that $(lastword $(MAKEFILE_LIST)) does not work if there is a space in the path name. so i'm not sure that there is a "perfect" way to do this.

---

> by the way... that (lastword(MAKEFILE_LIST)) does not work if there is a space in the path name. so i'm not sure that there is a "perfect" way to do this.

good point. But somebody on unix/linux having spaces in paths is calling for trouble. Stackoverflow also argues that [`make does not support filenames with whitespace in them.`](https://stackoverflow.com/questions/19153241/deal-with-makefile-list-spaces-on-windows).  `$(CURDIR)` would work for the `-C` case but alas would not work for the `-f` case.   I would tend to leave it as-is?  Definitely better than my original version and are we sure everything else works when there are spaces in paths?

---

one small request... we define DOCKER_DIR farther down in the Makefile. Can you move that up and set it instead of mkkfile_path (to be clear... it was ALREADY in the makefile, farther down, with the comment about why it was not being set with MAKEFILE_LIST). That being said... its set using MAKEFILE_LIST in the pdo-contracts repo.